packages/gateway-client/src/worker/update-app.ts

```diff
@@ -1,31 +1,423 @@
+import { levels } from 'loglevel';
+import { ClientMessageType } from '../worker-messaging';
 import { logger } from './logging';
+import messenger from './messenger';
+import { getVersion } from './version';
+
+declare const self: ServiceWorkerGlobalScope;
 
 const log = logger.getLogger('worker/update-app');
 
+// abort attempt to fetch updated script after 5 minutes
+const updateFetchTimeout = 5 * 60 * 1000;
```

Comment:
We'll need to decide how long this timeout should be. We don't want it to be too short, or else we'll be rolling back every time the user has a slow network. On the other hand, the user will have to keep the worker running for the full length of the timeout in order for the rollback to trigger.

Is 5 minutes too long? Will the user constantly give up after a few minutes and they'll just stay stuck with the bad update because the worker will never roll back? We're essentially telling people that if they want to get off of this bad update, they need to keep the client open for 5 minutes.